lib/rules/max-lines-per-function.js

```diff
@@ -157,6 +157,33 @@ module.exports = {
             return false;
         }
 
+        /**
+         * Search for the closest identifer for the node function.
+         * @param {ASTNode} node Node function
+         * @param {ASTNode} parent Parent node of the function
+         * @returns {ASTNode} The function identifer node or the function body if none exists
+         */
+        function reportedNode(node, parent) {
+
+            // function decleration name
+            if (node.id) {
+                return node.id;
+            }
+
+            // methods, getters and setters key
+            if (node.key) {
+                return node.key;
+            }
+
+            // function assigment variable
+            if (parent.id) {
+                return parent.id;
+            }
+
+            // inline function
+            return node;
+        }
```

Comment:
Here's a utility function that I think does the same thing. 

https://github.com/eslint/eslint/blob/main/lib/rules/utils/ast-utils.js#L1765